.github/workflows/codeql.yml

```diff
@@ -0,0 +1,57 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+name: 'CodeQL'
+
+on:
+  push:
+    branches: ['main']
+  pull_request:
+    branches: ['main']
+  schedule:
+    - cron: '0 0 * * 1'
```

Comment:
```suggestion
# A random time on a Tuesday once a week.
    - cron: '34 1 * * 2'
```

It's good practice not to have every cronjob use the same time each week as it leads to spikes in Datacenter use and is generally poor practice, so we avoid these sorts of times for that reason, every hour, every day at midnight etc.

Reply:
I've made the requested changes.